config/config.go

```diff
@@ -195,6 +205,73 @@ type Adapter struct {
 	Disabled bool `mapstructure:"disabled"`
 }
 
+// validateAdapterEndpoint validates to make sure that an adapter has a valid endpoint
+// associated with it
+func validateAdapterEndpoint(endpoint string, adapterName string, errs configErrors) configErrors {
+	if endpoint == "" {
+		errs = append(errs, fmt.Errorf("There's no default endpoint available for %s. Calls to this bidder/exchange will fail. "+
+			"Please set adapters.%s.endpoint in your app config", adapterName, adapterName))
+	} else {
+		// Check if endpoint has macros. If it does then resolve with dummy values
+		// for validation
+		if macros.MacroRegExp.MatchString(endpoint) {
+			endpointTemplate, err := template.New("endpointTemplate").Parse(endpoint)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Error parsing template: %s", err.Error()))
+				return errs
+			}
+			if endpoint, err = macros.ResolveMacros(*endpointTemplate, macros.EndpointTemplateParams{Host: dummyHost, PublisherID: dummyPublisherID}); err != nil {
```

Comment:
I believe it's better to use "_" rather than endpoint for the return value from ResolveMacros() in the above line (same comment in the next function, line 248).  ResolveMacros() currently returns the empty string when err in non-nil so it would seem the error messages printed (lines 224, 249) will not properly identify the bad argument.

Reply:
@josephveach Good catch! Actually, I was using `endpoint` there so that we get the resolved endpoint to validate in the next step (line 230). So, we can't replace it with `_` but I can hold that value in a temp variable and assign it to endpoint if err is nil.

Thanks for pointing it out!

Reply:
Yeah, somehow I didn't finish reading the two functions.  One further suggestion is to add more detail to the error messages (i.e., print both adapterName value and the offending string value) so that it's more clear exactly what failed.